addons/hr/models/hr_employee.py

```diff
@@ -519,6 +519,9 @@ def _get_calendar_attendances(self, date_from, date_to):
         self.ensure_one()
         return self.resource_calendar_id.get_work_duration_data(date_from, date_to)
 
+    def _get_work_entry_calendar(self, value=False):
```

Comment:
That's not really what I meant by generic solution... :sweat_smile: 
After discussing with Sofie, we will keep the method in the work entries modules and come up with a more generic solution afterwards to implement everywhere it needs correctly.
However, your initial solutions didn't really make sense since you were fetching employee information from the work entry through an argument rather than the work entry itself, which is confusing and overcomplicated.

Here's the best local solution that should be done in the PR:
- in `hr_work_entry/models/hr_work_entry.py`:
```py
def _get_calendar(self):
    return self.employee_id.resource_calendar_id or self.company_id.resource_calendar_id
``` 
We need a fallback on the company here because the employee calendar isn't required
- in `hr_work_entry_contract/models/hr_work_entry.py`:
```py
def _get_calendar(self):
    return self.contract_id.resource_calendar_id or super()._get_calendar()
```
Same here, the fallback is required if the contract has no calendar defined.

Then, from that we can call it from your method `_get_work_entries_data` by doing:
```py
work_entry_calendar = work_entries._get_calendar()
```
No need to get only one record of the `work_entries` recordset since the sorted dictionary already ensures that we will only get one calendar.

Reply:
Hello @Bertrand2 

I've implemented the changes
Apologies for the misunderstanding :pray: